### PR TITLE
Uno: Additions for SpacialGaze

### PR DESCRIPTION
**Pull request by Lord-Haji**
## Changelog
* UNO now shows hashcolors at most places
* Added Auto Stardust Distribution to UNO Winners
* Added UNO Boosts
* Added UNO Champion Badge Functionality

### CODE REVIEW

File: chat-plugins/uno.js [hunk 1]
```diff
@@ -152,12 +152,13 @@ class UNOgame extends Rooms.RoomGame {
 		this.room.add(msg).update();
 	}
 
-	getPlayers(showCards) {
+	getPlayers(showCards, hashColor) {
```
Comment (HoeenCoder) [inline chat-plugins/uno.js]: Why do we need the `hashColor` boolean anyways if its always going to use hashcolor?

File: chat-plugins/uno.js [hunk 2]
```diff
@@ -152,12 +152,13 @@ class UNOgame extends Rooms.RoomGame {
...
 		let playerList = Object.keys(this.players);
-		if (!showCards) {
-			return playerList.sort().map(id => Chat.escapeHTML(this.players[id].name));
+		if (!showCards && hashColor) {
+			return playerList.sort().map(id => SG.nameColor(this.players[id].name, true));
 		}
 		if (this.direction === -1) playerList = playerList.reverse();
+		if (hashColor) return playerList.map(id => `${(this.currentPlayer && this.currentPlayer === id ? "<strong>" : "")}${SG.nameColor(this.players[id].name, true)} (${this.players[id].hand.length}) ${(this.currentPlayer && this.currentPlayer === id ? "</strong>" : "")}`);
```
Comment (HoeenCoder) [inline chat-plugins/uno.js]: Dont auto-bold all names here, only 1 players name is bolded at a time here.

File: chat-plugins/uno.js [hunk 3]
```diff
@@ -468,13 +490,13 @@ exports.commands = {
 
 			room.game = new UNOgame(room, target);
 			room.add(`|uhtml|uno-${room.gameNumber}|<div class="broadcast-green"><p style="font-size: 14pt; text-align: center">A new game of <strong>UNO</strong> is starting!</p><p style="font-size: 9pt; text-align: center">Use <strong>/uno join</strong> to join the game.</p></div>`).update();
-			this.privateModCommand(`(A game of UNO was created by ${user.name}.)`);
+			this.privateModCommand(`(A game of UNO was created by ${SG.nameColor(user.name, true)}.)`);
```
Comment (HoeenCoder) [inline chat-plugins/uno.js]: On all `this.privateModCommand` and `this.addModCommand`, dont use hashcolor

File: chat-plugins/uno.js (hunk 1 above)
Comment (Lord-Haji) [reply chat-plugins/uno.js]: I PMd the reason :|

Comment (HoeenCoder) [reply chat-plugins/uno.js]: If were going to automatically show the color on usernames we don't really need to pass an extra variable do we? We can just automatically do it. And i posted this review before you pmed me on PS.

**Pull request merged**